lib/json_api/view.ex

```diff
@@ -0,0 +1,57 @@
+defmodule ElixirTools.JsonApi.View do
+  @moduledoc """
+  JSON API View is an overlay over the default Jaserializer view and provides functionality to apply
+  standards.
+  """
+
+  @type view :: String.t()
+
+  defmacro __using__(opts) do
+    alias ElixirTools.JsonApi
+
+    quote do
+      use JaSerializer, unquote(opts)
+
+      @spec render(String.t(), map) :: map
+      def render("index.json-api", data), do: JsonApi.View.render(__MODULE__, data)
+      def render("show.json-api", data), do: JsonApi.View.render(__MODULE__, data)
+      def render("errors.json-api", data), do: JaSerializer.PhoenixView.render_errors(data)
+
+      @spec format_datetime(DateTime.t()) :: String.t()
+      def format_datetime(value) do
+        value
+        |> Timex.Timezone.convert("UTC")
+        |> DateTime.to_iso8601()
+      end
+    end
+  end
+
+  @spec render(module, map) :: map
+  def render(serializer, data) do
+    serializer
+    |> JaSerializer.PhoenixView.render(data)
+    |> format()
+  end
+
+  @spec format(map) :: map
+  defp format(map) do
```

Comment:
I would suggest having 2 format functions.
The first one being
```
defp format(map) when is_map(map) do
  for {key, value} <- map, into: %{} do
      {convert_key(key), convert_value(value)}
  end
end
```
and second one
```
defp format(%{__struct__: _} = val) do
  val |> ensure_map |> format
end
```

I think this makes it a bit clearer than having a function parameter named `map` which might or might not be a map. This way will also I think fail with a clearer error message if a parameter of the wrong type is passed.
Thoughts on this?

Reply:
not related to this PR, in which I am just moving the existing logic to Elixir Tools. Feel free to refactor after though or by creating an issue: what you say makes some sense.